sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -0,0 +1,226 @@
+                # obfuscation errors are relatively common so only log them during debugging
+                self.log.debug("Failed to obfuscate query: %s", e)
+                obfuscated_statement = "ERROR: failed to obfuscate"
+            row = self._sanitize_row(row, obfuscated_statement)
+            estimated_size += self._get_estimated_row_size_bytes(row)
+            if estimated_size > max_bytes_limit:
+                # query results are ORDER BY transaction_begin_time ASC
+                # so once we hit the max bytes limit, return
+                return normalized_rows
+            normalized_rows.append(row)
+        return normalized_rows
+
+    @staticmethod
+    def _sanitize_row(row, obfuscated_statement):
+        row = {key: val for key, val in row.items() if key not in dm_exec_requests_exclude_keys and val is not None}
+        row['text'] = obfuscated_statement
+        if 'query_hash' in row:
+            row['query_hash'] = _hash_to_hex(row['query_hash'])
+        if 'query_plan_hash' in row:
+            row['query_plan_hash'] = _hash_to_hex(row['query_plan_hash'])
+        return row
+
+    @staticmethod
+    def _get_estimated_row_size_bytes(row):
+        return len(str(row))
+
+    def _create_activity_event(self, active_sessions, active_connections):
+        event = {
+            "host": self._db_hostname,
+            "ddagentversion": datadog_agent.get_version(),
+            "ddsource": "sqlserver",
+            "dbm_type": "activity",
+            "collection_interval": self.collection_interval,
+            "ddtags": self.check.tags,
+            "timestamp": time.time() * 1000,
+            "sqlserver_activity": active_sessions,
+            "sqlserver_connections": active_connections,
+        }
+        return event
+
+    def _truncate_activity_rows(self, rows, max_bytes):
+        pass
+
+    def collect_activity(self):
+        """
+        Collects all current activity for the SQLServer intance.
+        :return:
+        """
+        start_time = time.time()
+        try:
+            # re-use the check's conn module, but set extra_key=dbm-activity- to ensure we get our own
+            # raw connection. adodbapi and pyodbc modules are thread safe, but connections are not.
+            with self.check.connection.open_managed_default_connection(key_prefix=self._conn_key_prefix):
+                with self.check.connection.get_managed_cursor(key_prefix=self._conn_key_prefix) as cursor:
+                    connections = self._get_active_connections(cursor)
+                    rows = self._get_activity(cursor)
+                    normalized_rows = self._normalize_queries_and_filter_rows(rows, MAX_PAYLOAD_BYTES)
+                    event = self._create_activity_event(normalized_rows, connections)
+                    self._check.database_monitoring_query_activity(
+                        json.dumps(event, default=default_json_event_encoding)
+                    )
+        except Exception:
```

Comment:
`DBMAsyncJob` does standardized error handling for any unhandled exceptions in `run_job` so here it's best to let exceptions like this blow up all the way. See here https://github.com/DataDog/integrations-core/blob/4c77e76403ccb70bf6e09c55efc0bad11d6a66f0/datadog_checks_base/datadog_checks/base/utils/db/utils.py#L230-L269

And yes I just realized that I did also this wrong for statement metrics & plans 🤦  https://github.com/DataDog/integrations-core/blob/4dd0c5ef5c1f0b552192776a9888e3ec3d396b16/sqlserver/datadog_checks/sqlserver/statements.py#L257-L264

And we're actually doing it "wrong" in a couple of other places that need to be fixed.